File: SquidGameDalgonaFilter/Graph/ScriptTemplate/Select/CGSelect.js
/**
 * @file CGSelect.js
 * @author xuyuan
 * @date 2021/8/27
 * @brief CGSelect.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const APJS = require('../../../amazingpro');
const {BaseNode} = require('../Utils/BaseNode');

class CGSelect extends BaseNode {
  constructor() {
    super();
  }

  getOutput(index) {
    if (true === this.inputs[0]()) {
      return this.inputs[1]();
    } else {
      return this.inputs[2]();
    }
  }
}

exports.CGSelect = CGSelect;

File: SquidGameDalgonaFilter/Graph/ScriptTemplate/Select/CGSelect.lua
local CGSelect = CGSelect or {}
CGSelect.__index = CGSelect

function CGSelect.new()
    local self = setmetatable({}, CGSelect)
    self.inputs = {}
    return self
end

function CGSelect:setInput(index, func)
    self.inputs[index] = func
end

function CGSelect:getOutput(index)
    if self.inputs[0]() then
        return self.inputs[1]()
    else
        return self.inputs[2]()
    end
end

return CGSelect
